magma/array.py

```diff
@@ -275,6 +288,31 @@ def flat_length(cls):
         return cls.N * cls.T.flat_length()
 
     def __getitem__(self, key):
+        if isinstance(key, tuple):
+            if len(key) == 1:
+                return self[key[0]]
+            # ND Array key
+            if not (key[-1] == slice(None) or key[-1] == slice(0, None) or
```

Comment:
what is this condition? can you augment the comment to explain this a bit?

Reply:
Will leave a comment, it checks if the key is any of `x[:], x[0:], x[:len(x)], x[0:len(x)]`, basically checks if it's trying to slice the entire array